lib/core.js

```diff
@@ -475,7 +475,7 @@ declare class URIError extends Error {
 declare class JSON {
     static parse(text: string, reviver?: (key: any, value: any) => any): any;
     static stringify(
-      value: any,
+      value: null | string | number | boolean | Symbol | {} | $ReadOnlyArray<mixed>,
```

Comment:
A few thoughts:

- Interestingly, the spec doesn't mention `undefined`: https://www.ecma-international.org/ecma-262/5.1/#sec-15.12.3
- TypeScript allows `any`: https://github.com/Microsoft/TypeScript/blob/2f9218f346bbc606cccc4aebfca1c2c4f3167fee/lib/lib.es5.d.ts#L1042
- What do you do with custom types, like `class` instances?

```ts
JSON.stringify(undefined) // undefined

class A { constructor(){ this.b = 3 }}
JSON.stringify(new A) // '{"b":3}'
```

- A good compromise might be an overload:

```ts
declare class JSON {
  static stringify(value: typeof undefined): typeof undefined;
  static stringify(value: mixed): string;
}
```

Reply:
Replies to each point
1. This is why I didn't want to add the overload.
2. Typescript used any till now, however, https://github.com/Microsoft/TypeScript/pull/29744 made a PR there too. Also, https://twitter.com/fdsmars/status/1092634853378383872.
3. custom classes should be accounted for with `{}`.
4. An overload is unnecessary based on my points above. It's also redundant.

Reply:
FWIW, classes can provide custom behavior by implementing `toJSON()` method:

```js
class Foo {}
JSON.stringify(new Foo()) // "{}"

class Bar {
  toJSON() {
    return undefined;
  }
}
JSON.stringify(new Bar()) // undefined
```

Reply:
Good point to consider. I think this should still be okay since it accepts `{}`

Reply:
@stryju something like this? 
```js
declare function stringify<T: string | void>(
  object: { toJSON(): T }
): T
declare function stringify(
  object: null | string | number | boolean | Symbol | {} | $ReadOnlyArray<mixed>
): string
```